fabric-resource-loader-v0/src/main/java/net/fabricmc/fabric/api/resource/ResourceManagerHelper.java

```diff
@@ -52,4 +54,16 @@ default void addReloadListener(IdentifiableResourceReloadListener listener) {
 	static ResourceManagerHelper get(ResourceType type) {
 		return ResourceManagerHelperImpl.get(type);
 	}
+
+	/**
+	 * Registers a built-in resource pack.
+	 *
+	 * @param id The identifier of the resource pack.
+	 * @param subPath The sub path in the mod resources.
+	 * @param container The mod container.
+	 * @return True if successfully registered the resource pack, else false.
+	 */
+	static boolean registerBuiltinResourcePack(Identifier id, String subPath, ModContainer container) {
```

Comment:
I agree, hopefully the documentation I added is understandable/enough.

When a built-in resource pack is registered, there's a new resource pack which is by default disabled like in the test mod I added for this module:
![image](https://user-images.githubusercontent.com/12587332/90308723-2bf9fc80-dee2-11ea-9d29-2355100c6735.png)

It also applies to data packs.

It can be very useful for some mods even if it's a bit specific but I prefer it being directly in FAPI than copying the `ModNioResourcePack` code and all the injections into each mods using built-in resource packs.

For example if a mod add "snowy" variations to blocks it can just add the snow layer into the integrated mod resource pack so it doesn't break the style when paired with user resource packs, but if the user uses the Default resource pack for example, the mod can provide a built-in resource pack with snowy textures based on the Default textures and limit style issues.

It can also be useful for data packs which could provide extra features but for some reasons should be optional.

Reply:
Much better description